src/Diagnostics.DataProviders/DataProviderConfigurations/GeoMasterDataProviderConfiguration.cs

```diff
@@ -30,5 +30,15 @@ public GeoMasterDataProviderConfiguration()
         /// </summary>
         [ConfigurationName("GeoCertSubjectName")]
         public string GeoCertSubjectName { get; set; }
+        /// <summary>
+        /// Max Retry Count
+        /// </summary>
+        [ConfigurationName("Retry:MaxRetryCount")]
+        public int MaxRetryCount { get; set; }
```

Comment:
Hey @XiaoxuHU , I'd recommend setting these retry configuration settings at the base class `DataProviderConfigurationBase` so that you don't have to have duplicate code everywhere and less prone to future errors for these configuration values. https://github.com/Azure/Azure-AppServices-Diagnostics/blob/main/src/Diagnostics.DataProviders/DataProviderConfigurations/DataProviderConfigurationBase.cs

Reply:
Thank you, @hforeste ! I  updated code to move common settings into base configuration